dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -55,43 +65,44 @@ internal class DatadogRumMonitor(
             eventData,
             attributes
         )
-        activeView[key] = event
+        activeViewEvent = event
+        activeViewData = eventData
+        activeViewKey = WeakReference(key)
     }
 
     override fun stopView(
         key: Any,
         attributes: Map<String, Any?>
     ) {
-        val startedEvent = activeView.remove(key)
-        val startedEventData = startedEvent?.eventData as? RumEventData.View
+        val startedKey = activeViewKey.get()
+        val startedEvent = activeViewEvent
+        val startedEventData = activeViewData
 
         when {
-            startedEvent == null -> {
-                devLogger.w(
-                    "Unable to stop view with key <$key>. " +
-                        "This can mean that the view was not started or already stopped."
-                )
-            }
-            startedEventData == null -> {
+            startedEvent == null || startedEventData == null -> devLogger.w(
+                "Unable to end view with key <$key> (missing start). " +
+                    "This can mean that the view was not started or already ended."
+            )
+            startedKey == null -> {
                 devLogger.e(
-                    "Unable to stop view with key <$key>. " +
-                        "The related data was inconsistent."
+                    "Unable to end view with key <$key> (missing key)." +
+                        "Closing previous view automatically."
+                )
+                sendCompleteView(
+                    startedEvent,
+                    startedEventData,
+                    mapOf(RumEventSerializer.TAG_EVENT_UNSTOPPED to true)
                 )
             }
-            else -> {
-                val updatedDurationNs = System.nanoTime() - startedEventData.durationNanoSeconds
-                val updatedEvent = startedEvent.copy(
-                    eventData = startedEventData.copy(
-                        durationNanoSeconds = updatedDurationNs,
-                        version = startedEventData.version + 1
-                    ),
-                    attributes = startedEvent.attributes.toMutableMap().apply {
-                        putAll(attributes)
-                    }
+            startedKey != key -> {
+                devLogger.e(
+                    "Unable to end view with key <$key> (mismatched key). " +
+                        "Another view with key <$startedKey> has been started."
                 )
-                writer.write(updatedEvent)
             }
+            else -> sendCompleteView(startedEvent, startedEventData, attributes)
```

Comment:
you do not want to send the unclosed resources here also ?

Reply:
That's a good idea